tests/cpp/test_gpu_view.cpp

```diff
@@ -2619,4 +2620,87 @@ TEST_F(GpuViewTest, GroupNorm) {
   testValidate(
       executor_cache.fusion(), cg_outputs, {t0}, {ref}, __LINE__, __FILE__);
 }
+
+TEST_F(GpuViewTest, FusionMismatchingReshape) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  // TODO: use symbolic sizes. Currently, this is not working because of
+  // failures in replaceSymbolicSizes
+  TensorView* tv0 = makeContigConcreteTensor({2, 3, 5}, DataType::Float);
+  // TensorView* tv0 = makeContigTensor(3, DataType::Float);
+  fusion.addInput(tv0);
+  auto tv1 = sin(tv0);
+  auto tv2 = reshape(tv1, {2, 3, 5}, {2 * 3, 5});
+  auto tv3 = reshape(tv1, {2, 3, 5}, {2, 3 * 5});
+  auto tv4 = cos(tv2);
+  auto tv5 = exp(tv3);
+  fusion.addOutput(tv4);
+  fusion.addOutput(tv5);
+
```

Comment:
Could you give some general description of the scheduling approach? I'm lost at line 2649.

Reply:
Added